### PR TITLE
small cleanup

### PR DESCRIPTION
**Pull request by Kukks**
Just a small subset of #850

### CODE REVIEW

File: BTCPayServer/Hosting/BTCPayServerServices.cs [hunk 1]
```diff
@@ -177,15 +179,18 @@ public static IServiceCollection AddBTCPayServer(this IServiceCollection service
...
+            services.AddSingleton<BitcoinLikePaymentHandler>();
+            services.AddSingleton<IPaymentMethodHandler>(provider => provider.GetService<BitcoinLikePaymentHandler>());
+            services.AddSingleton<IPaymentMethodHandler<DerivationSchemeSettings>>(provider => provider.GetService<BitcoinLikePaymentHandler>());
+            services.AddSingleton<IHostedService, NBXplorerListener>();
 
-            services.AddSingleton<Payments.IPaymentMethodHandler<Payments.Lightning.LightningSupportedPaymentMethod>, Payments.Lightning.LightningLikePaymentHandler>();
             services.AddSingleton<LightningLikePaymentHandler>();
-            services.AddSingleton<IHostedService, Payments.Lightning.LightningListener>();
-            
+            services.AddSingleton<IPaymentMethodHandler>(provider => provider.GetService<LightningLikePaymentHandler>());
```
Comment (NicolasDorier) [inline BTCPayServer/Hosting/BTCPayServerServices.cs]: why??

Comment (Kukks) [reply BTCPayServer/Hosting/BTCPayServerServices.cs]: Before, we were creating the duplicate instances of the same service( under `LightningLikePaymentHandler`  and `IPaymentMethodHandler<LightningSupportedPaymentMethod>`). 
This makes them all use the same one while also registering them against the main interface `IPaymentMethodHandler`

This is for my next pull requests (starting at #852 )

Comment (NicolasDorier) [reply BTCPayServer/Hosting/BTCPayServerServices.cs]: If there is several singleton using `IPaymentMethodHandler`, then controllers can resolve `IEnumerable<IPaymentMethodHandler>` ?

Comment (Kukks) [reply BTCPayServer/Hosting/BTCPayServerServices.cs]: Yes

**Pull request merged**